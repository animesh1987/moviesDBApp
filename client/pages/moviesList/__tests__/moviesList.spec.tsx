import * as React from 'react';
import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';

import { MoviesList } from '../index';

describe('Movies List Test', () => {
  const movie = {
    id: 1,
    title: 'Test',
    overview: 'Test overview',
    release_date: "20-04-04",
    runtime: 144,
    poster_path: 'abc',
    genres: ['test']
  };

  const testFn = jest.fn();

  it('Empty state renders correctly', () => {
    const component = create(
      <MoviesList
        goToMovie={testFn}
        movies={[]}/>)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('Movie renders correctly', () => {
    const component = create(
      <MoviesList
        goToMovie={testFn}
        movies={[movie]}/>)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('gotomovie is called correctly', () => {
    const component = shallow(
      <MoviesList
        goToMovie={testFn}
        movies={[movie]}/>
    );
    console.log(component.childAt(1).html());
    expect(true).toBe(true);
  });
});
