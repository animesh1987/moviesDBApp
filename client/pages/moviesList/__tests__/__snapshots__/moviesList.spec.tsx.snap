// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies List Test Empty state renders correctly 1`] = `
<div
  className="empty-state"
>
  <h1>
    No films found
  </h1>
</div>
`;

exports[`Movies List Test Movie renders correctly 1`] = `
<div
  className="layout-row movies-container"
>
  <div
    className="movie-card"
    onClick={[Function]}
  >
    <div
      className="banner-image"
      style={
        Object {
          "backgroundImage": "url(abc)",
        }
      }
    />
    <div
      className="layout-row layout-align-space-between layout-align-center"
    >
      <span
        className="movie-title"
      >
        Test
      </span>
      <span
        className="release-date"
      >
        20-0
      </span>
    </div>
    <div
      className="genre-details"
    >
      test
    </div>
  </div>
</div>
`;
